app/controllers/authentication_controller.rb

```diff
@@ -24,8 +24,13 @@ def callback_from_omniauth
   end
 
   def sign_out
-    warden.logout
-    redirect_to send(:"#{params[:provider]}_sign_out_url"), allow_other_host: true
+    # get current_user before user is logged out of warden
+    if (auth_provider = current_user&.provider)
```

Comment:
Yes, it is deliberate, I was worrying when I wrote it that it might be a bit ambiguous... I'll split up the assignment and the test to make it clearer.

Reply:
I've changed things slightly, what do you think? Better or worse? 😅